bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -371,6 +371,11 @@ func (r *KubeadmConfigReconciler) refreshBootstrapTokenIfNeeded(ctx context.Cont
 
 	secret, err := getToken(ctx, remoteClient, token)
 	if err != nil {
+		if apierrors.IsNotFound(err) && scope.ConfigOwner.IsMachinePool() {
+			log.Info("Bootstrap token secret is gone, triggering creation of new token")
```

Comment:
nit
```suggestion
			log.Info("Bootstrap token secret is not found, triggering creation of new token")
```
same in L410

Reply:
Might be better to move this log line and the following line blanking out the Token into recreateBootstrapToken

q: can we drop blanking out the boolstrap token entirely, given that we are setting it to a new value inside recreateBootstrapToken

Reply:
Reworded.

Semantically, the function name `recreateBootstrapToken` doesn't denote that the secret wasn't found, so I'd rather leave the log line _here_ where we determined that the secret wasn't found and _therefore_ need to call the recreation functionality.

We could drop setting the field to `""`, but then if token creation fails, CAPI tries to read the non-existing secret again. If we set an empty string, the next reconciliation run will try to create the token. Your call here – I have no strong opinion which way to go.